bokeh/models/sources.py

```diff
@@ -132,6 +139,8 @@ def __init__(self, *args, **kw):
                 raw_data = self._data_from_groupby(raw_data)
             else:
                 raise ValueError("expected a dict or pandas.DataFrame, got %s" % raw_data)
+        else:
+            raw_data = copy.copy(raw_data)
```

Comment:
This makes copies of plain `dict` as well, which may slightly change the semantics (and is not really necessary). I wonder if it would be better to  check specifically for `PropertyValueColumnData` instances, and only call `copy` in that case?

Reply:
I though that this would at least be consistent as it just makes copies of any ``dict`` subclass passed.
I can implement the typecheck of course, but maybe I could also just remove these two lines altogether, as the functionality I requested is covered by the ``__deepcopy__`` hook and in my opinion making a shallow copy only has limited use cases anyway. Then I would slightly change the docstring ``.. note::`` wording to inform users about the need to invoke ``deepcopy()`` on the data object should they require an independent copy.

I'm fine either way, though this is all the time I have today, so I will implement the tests and any other changes another time.